python/upt-cpan/Portfile

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                upt-cpan
+version             0.4
+revision            0
+
+maintainers         nomaintainer
+description         CPAN frontend for upt
+long_description    ${description}
+
+platforms           darwin
+supported_archs     noarch
+
+homepage            https://framagit.org/upt/upt-cpan
+master_sites        pypi:[string index ${python.rootname} 0]/${python.rootname}
+distname            ${python.rootname}-${version}
+
+license             3-clause BSD License
+checksums           sha256  47ebc37bf4510867858eec52e2d23221bc5a0f061a057b41892cb96b8e3def14 \
+                    rmd160  418532c5abb388d4ddcb050610944266ba1f456c \
+                    size    5091
+                                
+python.versions     37
+
+
```

Comment:
the code below needs to be in a ```if {${name} ne ${subport}} {``` block

Reply:
Before it was in a `if {${name} ne ${subport}} {` block but while running `port -vst install`, error occurred so I had to change to this.
ref: https://lists.macports.org/pipermail/macports-dev/2019-April/040522.html

Reply:
yes, I did read that message. However, I don't see why that wouldn't work as this is the way it's done for almost all other Python ports. If there is a reason why this doesn't work, we should probably fix that problem.

Reply:
After Changing the name to `py-upt`, it works with the if block.

I will update the PR with new commits with all the changes pointed out here.

Reply:
Cool, I am glad it works now! It looks like the Python portgroup assumes (correctly) that a port in the "python" directory starts with ```py-```. If that's not the case some (or all) of the defaults the portfile normally sets, does not work.

Reply:
No, it uses the name to decide whether the port is a python module or an app that uses python. This is documented: https://guide.macports.org/chunked/reference.portgroup.html#reference.portgroup.python